### PR TITLE
Add definition for logging.metrics.enabled, pass config from agent to beats

### LINKED ISSUE #26758: [Elastic Agent] Sending monitoring metrics to logs datastream
It looks like Elastic Agent is sending it's own monitoring metrics to the logs datastream. This is not useful as a log message. It should be sending this information to the metrics datastream instead. 

Example event:
```
{
  "_index": ".ds-logs-elastic_agent.metricbeat-default-2021.07.07-000196",
  "_type": "_doc",
  "_id": "DWZtgXoBysd6RRjof1WI",
  "_version": 1,
  "_score": null,
  "fields": {
    "elastic_agent.version": [
      "7.13.2"
    ],
    "monitoring.metrics.beat.cpu.system.ticks": [
      2191340
    ],
    "monitoring.metrics.metricbeat.system.socket.events": [
      8
    ],
    "monitoring.metrics.metricbeat.system.cpu.events": [
      3
    ],
    "host.hostname": [
      "unifi"
    ],
    "host.mac": [
    ],
    "monitoring.metrics.metricbeat.system.network_summary.events": [
      3
    ],
    "monitoring.metrics.libbeat.output.write.bytes": [
      382478
    ],
    "monitoring.metrics.metricbeat.system.network_summary.success": [
      3
    ],
    "host.os.version": [
      "18.04.5 LTS (Bionic Beaver)"
    ],
    "monitoring.metrics.metricbeat.system.service.events": [
      162
    ],
    "agent.name": [
      "unifi"
    ],
    "monitoring.metrics.beat.info.uptime.ms": [
      162121222
    ],
    "monitoring.metrics.metricbeat.system.service.success": [
      162
    ],
    "monitoring.metrics.beat.memstats.memory_alloc": [
      19120104
    ],
    "host.os.type": [
      "linux"
    ],
    "monitoring.metrics.metricbeat.system.entropy.success": [
      3
    ],
    "monitoring.metrics.metricbeat.system.raid.failures": [
      3
    ],
    "monitoring.metrics.metricbeat.system.uptime.success": [
      3
    ],
    "input.type": [
      "log"
    ],
    "monitoring.metrics.libbeat.pipeline.clients": [
      17
    ],
    "agent.hostname": [
      "unifi"
    ],
    "monitoring.metrics.libbeat.pipeline.events.total": [
      248
    ],
    "monitoring.metrics.libbeat.pipeline.events.active": [
      0
    ],
    "host.architecture": [
      "x86_64"
    ],
    "monitoring.metrics.metricbeat.system.socket.success": [
      8
    ],
    "agent.id": [
      "0d376a06-dccc-4fb9-94ea-d8c5ecea380a"
    ],
    "host.containerized": [
      false
    ],
    "monitoring.metrics.system.load.norm.15": [
      0.11
    ],
    "monitoring.metrics.beat.cpu.total.value": [
      4682010
    ],
    "monitoring.metrics.metricbeat.system.load.events": [
      3
    ],
    "log.logger": [
      "monitoring"
    ],
    "monitoring.metrics.libbeat.pipeline.queue.acked": [
      248
    ],
    "host.ip": [
    ],
    "agent.type": [
      "filebeat"
    ],
    "monitoring.metrics.metricbeat.system.diskio.events": [
      21
    ],
    "monitoring.metrics.beat.handles.open": [
      19
    ],
    "monitoring.metrics.metricbeat.system.diskio.success": [
      21
    ],
    "monitoring.metrics.beat.cpu.total.ticks": [
      4682010
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "host.id": [
      "a4719423bdb94e1e80df8ea652c5dd59"
    ],
    "monitoring.metrics.libbeat.output.events.active": [
      0
    ],
    "monitoring.metrics.system.load.5": [
      0.12
    ],
    "monitoring.metrics.beat.memstats.memory_total": [
      613619360744
    ],
    "elastic_agent.id": [
      "3cace087-149e-4507-b0ee-5e6a3afdf14a"
    ],
    "monitoring.metrics.metricbeat.system.process.success": [
      18
    ],
    "host.os.codename": [
      "bionic"
    ],
    "monitoring.metrics.system.load.1": [
      0.27
    ],
    "monitoring.metrics.beat.memstats.rss": [
      39579648
    ],
    "monitoring.metrics.metricbeat.system.load.success": [
      3
    ],
    "log.origin.file.name": [
      "log/log.go"
    ],
    "@timestamp": [
      "2021-07-07T14:44:33.126Z"
    ],
    "host.os.platform": [
      "ubuntu"
    ],
    "log.file.path": [
      "/opt/Elastic/Agent/data/elastic-agent-686ba4/logs/default/metricbeat-json.log"
    ],
    "data_stream.dataset": [
      "elastic_agent.metricbeat"
    ],
    "agent.ephemeral_id": [
      "8361cc5c-d090-45a6-a564-3ccc134c261b"
    ],
    "monitoring.metrics.metricbeat.system.memory.success": [
      3
    ],
    "monitoring.metrics.metricbeat.system.uptime.events": [
      3
    ],
    "monitoring.metrics.metricbeat.system.network.events": [
      12
    ],
    "monitoring.metrics.metricbeat.system.cpu.success": [
      3
    ],
    "monitoring.metrics.libbeat.pipeline.events.published": [
      248
    ],
    "monitoring.metrics.beat.memstats.gc_next": [
      21159136
    ],
    "monitoring.metrics.metricbeat.system.process_summary.success": [
      3
    ],
    "host.os.name": [
      "Ubuntu"
    ],
    "log.level": [
      "info"
    ],
    "monitoring.metrics.beat.runtime.goroutines": [
      112
    ],
    "host.name": [
      "unifi"
    ],
    "monitoring.metrics.beat.cpu.system.time.ms": [
      376
    ],
    "monitoring.metrics.metricbeat.system.socket_summary.success": [
      3
    ],
    "monitoring.metrics.metricbeat.system.network.success": [
      12
    ],
    "log.offset": [
      220314
    ],
    "data_stream.type": [
      "logs"
    ],
    "monitoring.metrics.libbeat.output.events.total": [
      248
    ],
    "monitoring.metrics.beat.handles.limit.soft": [
      1024
    ],
    "ecs.version": [
      "1.8.0"
    ],
    "agent.version": [
      "7.13.2"
    ],
    "monitoring.metrics.libbeat.output.events.batches": [
      9
    ],
    "host.os.family": [
      "debian"
    ],
    "monitoring.metrics.metricbeat.system.raid.events": [
      3
    ],
    "monitoring.metrics.metricbeat.system.entropy.events": [
      3
    ],
    "monitoring.metrics.beat.cpu.user.time.ms": [
      490
    ],
    "monitoring.metrics.beat.cgroup.memory.mem.usage.bytes": [
      81920
    ],
    "monitoring.metrics.beat.cgroup.cpuacct.total.ns": [
      1159162390
    ],
    "monitoring.metrics.system.load.norm.1": [
      0.27
    ],
    "monitoring.metrics.system.load.norm.5": [
      0.12
    ],
    "monitoring.metrics.beat.cpu.user.ticks": [
      2490670
    ],
    "monitoring.ecs.version": [
      "1.6.0"
    ],
    "monitoring.metrics.metricbeat.system.process_summary.events": [
      3
    ],
    "monitoring.metrics.metricbeat.system.memory.events": [
      3
    ],
    "monitoring.metrics.libbeat.output.read.bytes": [
      62838
    ],
    "monitoring.metrics.system.load.15": [
      0.11
    ],
    "monitoring.metrics.beat.cpu.total.time.ms": [
      866
    ],
    "monitoring.metrics.beat.info.ephemeral_id": [
      "c8914934-42d8-4d7a-9ae5-bda5db39d2e0"
    ],
    "monitoring.metrics.libbeat.config.module.running": [
      17
    ],
    "host.os.kernel": [
      "4.15.0-147-generic"
    ],
    "monitoring.metrics.libbeat.output.events.acked": [
      248
    ],
    "monitoring.metrics.metricbeat.system.process.events": [
      18
    ],
    "log.origin.file.line": [
      144
    ],
    "monitoring.metrics.metricbeat.system.socket_summary.events": [
      3
    ],
    "data_stream.namespace": [
      "default"
    ],
    "message": [
      "Non-zero metrics in the last 30s"
    ],
    "monitoring.metrics.beat.handles.limit.hard": [
      4096
    ],
    "event.dataset": [
      "elastic_agent.metricbeat"
    ]
  },
  "sort": [
    1625669073126
  ]
}
```

For confirmed bugs, please report:
- Version: 7.13.2
- Operating System: Linux
- Steps to Reproduce: Install Elastic Agent and enable metrics monitoring in the agent policy

### PR DESCRIPTION
**Pull request by michel-laterman**
## What does this PR do?

Define logging.metrics.enabled and logging.metrics.period.
When logging.metrics.enabled is set to false the elastic-agent will
pass the setting to the underlying beats through a command line variable.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] ~~I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the [documentation](https://github.com/elastic/observability-docs/pull/853)
- [x] I have made corresponding change to the default configuration files
- [x] ~~I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Add `agent.monitoring.log_metrics: false` to elastic-agent.yml (in stand-alone mode) and ensure that the beat logs do not have metrics entries.

i.e.: default behaviour with agent.monitoring.log_metrics missing:
```
elastic-agent-7.14.0-SNAPSHOT-darwin-x86_64|⇒ ps aux | grep beat
laterman          4742   0.0  0.0  4408696    828 s000  R+    9:50am   0:00.00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox beat
root              4645   0.0  0.1  5747796  82824   ??  S     9:49am   0:00.49 /Library/Elastic/Agent/data/elastic-agent-6e0888/install/metricbeat-7.14.0-SNAPSHOT-darwin-x86_64/metricbeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${METRICBEAT_GOGC:100} -E logging.level=info -E http.enabled=true -E http.host=unix:///Library/Elastic/Agent/data/tmp/default/metricbeat/metricbeat.sock_monitor -E logging.json=true -E logging.ecs=true -E logging.files.path=/Library/Elastic/Agent/data/elastic-agent-6e0888/logs/default -E logging.files.name=metricbeat_monitor-json.log -E logging.files.keepfiles=7 -E logging.files.permission=0640 -E logging.files.interval=1h -E path.data=/Library/Elastic/Agent/data/elastic-agent-6e0888/run/default/metricbeat--7.14.0-SNAPSHOT--36643631373035623733363936343635
root              4644   0.0  0.1  6020232  68824   ??  S     9:49am   0:00.41 /Library/Elastic/Agent/data/elastic-agent-6e0888/install/filebeat-7.14.0-SNAPSHOT-darwin-x86_64/filebeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${FILEBEAT_GOGC:100} -E logging.level=info -E http.enabled=true -E http.host=unix:///Library/Elastic/Agent/data/tmp/default/filebeat/filebeat.sock_monitor -E logging.json=true -E logging.ecs=true -E logging.files.path=/Library/Elastic/Agent/data/elastic-agent-6e0888/logs/default -E logging.files.name=filebeat_monitor-json.log -E logging.files.keepfiles=7 -E logging.files.permission=0640 -E logging.files.interval=1h -E path.data=/Library/Elastic/Agent/data/elastic-agent-6e0888/run/default/filebeat--7.14.0-SNAPSHOT--36643631373035623733363936343635
root              4643   0.0  0.1  5808224 100492   ??  S     9:49am   0:06.82 /Library/Elastic/Agent/data/elastic-agent-6e0888/install/metricbeat-7.14.0-SNAPSHOT-darwin-x86_64/metricbeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${METRICBEAT_GOGC:100} -E logging.level=info -E http.enabled=true -E http.host=unix:///Library/Elastic/Agent/data/tmp/default/metricbeat/metricbeat.sock -E logging.json=true -E logging.ecs=true -E logging.files.path=/Library/Elastic/Agent/data/elastic-agent-6e0888/logs/default -E logging.files.name=metricbeat-json.log -E logging.files.keepfiles=7 -E logging.files.permission=0640 -E logging.files.interval=1h -E path.data=/Library/Elastic/Agent/data/elastic-agent-6e0888/run/default/metricbeat--7.14.0-SNAPSHOT
root              4641   0.0  0.1  5184168  68088   ??  S     9:48am   0:00.43 /Library/Elastic/Agent/data/elastic-agent-6e0888/install/filebeat-7.14.0-SNAPSHOT-darwin-x86_64/filebeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${FILEBEAT_GOGC:100} -E logging.level=info -E http.enabled=true -E http.host=unix:///Library/Elastic/Agent/data/tmp/default/filebeat/filebeat.sock -E logging.json=true -E logging.ecs=true -E logging.files.path=/Library/Elastic/Agent/data/elastic-agent-6e0888/logs/default -E logging.files.name=filebeat-json.log -E logging.files.keepfiles=7 -E logging.files.permission=0640 -E logging.files.interval=1h -E path.data=/Library/Elastic/Agent/data/elastic-agent-6e0888/run/default/filebeat--7.14.0-SNAPSHOT
```

setting log_metrics: false:
```
elastic-agent-7.14.0-SNAPSHOT-darwin-x86_64|⇒ ps aux | grep beat
root              4816   8.5  0.1  5941648  94980   ??  S     9:51am   0:01.27 /Library/Elastic/Agent/data/elastic-agent-6e0888/install/metricbeat-7.14.0-SNAPSHOT-darwin-x86_64/metricbeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${METRICBEAT_GOGC:100} -E logging.level=info -E http.enabled=true -E http.host=unix:///Library/Elastic/Agent/data/tmp/default/metricbeat/metricbeat.sock -E logging.json=true -E logging.ecs=true -E logging.files.path=/Library/Elastic/Agent/data/elastic-agent-6e0888/logs/default -E logging.files.name=metricbeat-json.log -E logging.files.keepfiles=7 -E logging.files.permission=0640 -E logging.files.interval=1h -E logging.metrics.enabled=false -E path.data=/Library/Elastic/Agent/data/elastic-agent-6e0888/run/default/metricbeat--7.14.0-SNAPSHOT
root              4825   3.7  0.1  5132456  61964   ??  S     9:51am   0:00.28 /Library/Elastic/Agent/data/elastic-agent-6e0888/install/filebeat-7.14.0-SNAPSHOT-darwin-x86_64/filebeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${FILEBEAT_GOGC:100} -E logging.level=info -E http.enabled=true -E http.host=unix:///Library/Elastic/Agent/data/tmp/default/filebeat/filebeat.sock_monitor -E logging.json=true -E logging.ecs=true -E logging.files.path=/Library/Elastic/Agent/data/elastic-agent-6e0888/logs/default -E logging.files.name=filebeat_monitor-json.log -E logging.files.keepfiles=7 -E logging.files.permission=0640 -E logging.files.interval=1h -E logging.metrics.enabled=false -E path.data=/Library/Elastic/Agent/data/elastic-agent-6e0888/run/default/filebeat--7.14.0-SNAPSHOT--36643631373035623733363936343635
laterman          4829   0.0  0.0  4408696    832 s000  R+    9:51am   0:00.00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox beat
root              4826   0.0  0.1  5139548  75556   ??  S     9:51am   0:00.26 /Library/Elastic/Agent/data/elastic-agent-6e0888/install/metricbeat-7.14.0-SNAPSHOT-darwin-x86_64/metricbeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${METRICBEAT_GOGC:100} -E logging.level=info -E http.enabled=true -E http.host=unix:///Library/Elastic/Agent/data/tmp/default/metricbeat/metricbeat.sock_monitor -E logging.json=true -E logging.ecs=true -E logging.files.path=/Library/Elastic/Agent/data/elastic-agent-6e0888/logs/default -E logging.files.name=metricbeat_monitor-json.log -E logging.files.keepfiles=7 -E logging.files.permission=0640 -E logging.files.interval=1h -E logging.metrics.enabled=false -E path.data=/Library/Elastic/Agent/data/elastic-agent-6e0888/run/default/metricbeat--7.14.0-SNAPSHOT--36643631373035623733363936343635
root              4818   0.0  0.1  5137808  59964   ??  S     9:51am   0:00.24 /Library/Elastic/Agent/data/elastic-agent-6e0888/install/filebeat-7.14.0-SNAPSHOT-darwin-x86_64/filebeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${FILEBEAT_GOGC:100} -E logging.level=info -E http.enabled=true -E http.host=unix:///Library/Elastic/Agent/data/tmp/default/filebeat/filebeat.sock -E logging.json=true -E logging.ecs=true -E logging.files.path=/Library/Elastic/Agent/data/elastic-agent-6e0888/logs/default -E logging.files.name=filebeat-json.log -E logging.files.keepfiles=7 -E logging.files.permission=0640 -E logging.files.interval=1h -E logging.metrics.enabled=false -E path.data=/Library/Elastic/Agent/data/elastic-agent-6e0888/run/default/filebeat--7.14.0-SNAPSHOT
```

## Related issues

- Closes #26758

### CODE REVIEW

**Comment by ruflin** [comment]
Can you share your thoughts on if we should set the default to true (as it is) or false?

**Comment by michel-laterman** [comment]
@urso and I briefly discussed this, setting it to true does not change existing behaviour and allows us to use the logs when providing support to users.

File: x-pack/elastic-agent/_meta/config/common.p2.yml.tmpl [hunk 1]
```diff
@@ -33,6 +33,8 @@ inputs:
 #   logs: true
 #   # enables metrics monitoring
 #   metrics: true
+#   # running processes will emit metrics into logs
+#   log_metrics: true
```
Comment (urso) [inline x-pack/elastic-agent/_meta/config/common.p2.yml.tmpl:37]: What is the difference between `metrics` and `logs_metrics`? 

We already have the `agent.logging.metrics.enabled` settings. Should we rather use that setting?

@michalpristas WDYT?

Comment (michel-laterman) [reply x-pack/elastic-agent/_meta/config/common.p2.yml.tmpl:37]: need to check if these settings are passed to beats

Comment (michalpristas) [reply x-pack/elastic-agent/_meta/config/common.p2.yml.tmpl:37]: i'm not against of reusing what we have. these settings are not forwarded to beats but used for constructing our logger.

**Comment by michalpristas** [comment]
where are we on this. any updates?

File: libbeat/logp/config.go [hunk 2]
```diff
@@ -59,6 +60,14 @@ type FileConfig struct {
 	RedirectStderr  bool            `config:"redirect_stderr" yaml:"redirect_stderr"`
 }
 
+// MetricsConfig contains configuration used by the monitor to output metrics into the logstream.
+//
+// Currently these options are not used through this object in beats (as monitoring is setup elsewhere).
+type MetricsConfig struct {
+	Enabled bool          `config:"enabled" yaml:"enabled"`
```
Comment (michel-laterman) [inline libbeat/logp/config.go:67]: @urso, I noticed that the beats load through https://github.com/elastic/beats/blob/60e9e348b9aef83e3addf997a5efede05821fa23/libbeat/cmd/instance/beat.go#L453

but the logging config does define these variables, do you think it makes sense to define it as part of the logging config so it can be retrieved by the agent?

Comment (urso) [reply libbeat/logp/config.go:67]: +1

File: libbeat/tests/system/config/mockbeat.yml.j2 [hunk 3]
```diff
@@ -98,8 +98,8 @@ setup.ilm:
 
 #================================ Logging =====================================
 
-{% if metrics_period -%}
-logging.metrics.period: {{ metrics_period }}
+{% if file_name -%}
+logging.files.name: {{ file_name }}
```
Comment (michel-laterman) [inline libbeat/tests/system/config/mockbeat.yml.j2:102]: The export command being tested disables environment variable resolution.
https://github.com/elastic/beats/blob/4727470f7f7c3fd31c4b3c9f82c77beb8fe9b070/libbeat/cmd/export/config.go#L42
I think it's so `export config` can be used to copy the config and ignore the local environment.
Originally the test was for `logging.metrics.period` but seeing as how this pr adds the definition for this attribute as a `time.Duration` the export fails because the string could not be parsed as a duration.
I instead changed the test to use a string type attribute so the export does not fail.

However going forward the `export config` command will run into this issue if `logging.metrics.period` or `logging.metrics.enabled` are set to environment variables.
This should not effect other commands (which resolve environment variables).

tldr; attribute+test has changed because environment variable expansion is disabled for the `export` commands
